lib/rest-adapter.js

```diff
@@ -619,19 +620,49 @@ RestMethod.prototype.acceptsSingleBodyArgument = function() {
     getTypeString(accepts.type) == 'object' || false;
 };
 
+RestMethod.prototype.getEndpoints = function() {
+  var self = this;
+  return this.routes.map(function(route) {
+    var verbResult;
+    var verb = route.verb;
+    if (verb == 'all') {
+      verbResult = 'POST';
+    } else if (verb == 'del') {
+      verbResult = 'DELETE';
+    } else {
+      verbResult = verb.toUpperCase();
+    }
+    return {
+      verb: verbResult,
+      fullPath: joinPaths(self.restClass.getPath(), route.path),
+    };
+  });
+};
+
 RestMethod.prototype.getHttpMethod = function() {
-  var verb = this.routes[0].verb;
-  if (verb == 'all') return 'POST';
-  if (verb == 'del') return 'DELETE';
-  return verb.toUpperCase();
+  // depricate message to let the users know what they were using
+  // was retuning just the first route's verb
+  deprecated('getHttpMethod() is deprecated, use getFirstHttpMethod() instead.');
+  return this.getFirstHttpMethod();
+};
+
+RestMethod.prototype.getFirstHttpMethod = function() {
```

Comment:
On the second thought, I think `getFirstHttpMethod` may be not worth the API pollution. Users can always call `this.getEndpoints()[0].verb` themselves. What do you think?